datastore.go

```diff
@@ -130,6 +131,47 @@ func DiskUsage(d Datastore) (uint64, error) {
 	return persDs.DiskUsage()
 }
 
+// TTLDatastore is an interface that should be implemented by datastores that
+// support expiring entries.
+type TTLDatastore interface {
+	Datastore
+
+	PutWithTTL(key Key, value interface{}, ttl time.Duration) error
+	SetTTL(key Key, ttl time.Duration) error
+}
+
+// Txn is an interface for transactions that can be committed or discarded.
+type Txn interface {
+	// Txn has the same capabilities as Datastore. Txns allow users to batch
+	// queries and mutations to the Datastore into atomic groups, or transactions.
+	// Actions performed on a transaction will not take hold until a successful
+	// call to Commit has been made. Likewise, transactions can be aborted by
+	// calling Discard before a successful Commit has been made.
```

Comment:
This comment should be above the `type Txn interface` line.

Reply:
Yes, and maybe this could be named `TxnDatastore`.